lib/AST/Decl.cpp

```diff
@@ -2584,6 +2584,13 @@ ObjCClassKind ClassDecl::checkObjCAncestry() const {
   return kind;
 }
 
+ClassDecl::MetaclassKind ClassDecl::getMetaclassKind() const {
+  assert(getASTContext().LangOpts.EnableObjCInterop &&
+         "querying metaclass kind without objc interop");
+  auto objc = checkObjCAncestry() != ObjCClassKind::NonObjC || hasClangNode();
```

Comment:
Is it ever the case that checkObjCAncestry() == NonObjC, but hasClangNode() is true?